pkg/pipeline/stream/logger.go

```diff
@@ -21,20 +21,39 @@ import (
 )
 
 // SetLogger figures out if the node needs a logger, sets static metadata in the
-// logger and supplies it to the node.
-func SetLogger(n Node, logger log.CtxLogger) {
+// logger and supplies it to the node. Behavior can be customized by supplying
+// custom options.
+func SetLogger(n Node, logger log.CtxLogger, options ...func(log.CtxLogger, Node) log.CtxLogger) {
```

Comment:
This function is backwards compatible with existing calls but adds the ability to change the behavior if needed (we want to change the behavior for parallel worker nodes).